Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp

```diff
@@ -210,7 +210,9 @@ void MediaRecorderPrivateBackend::fetchData(MediaRecorderPrivate::FetchDataCallb
         Locker locker { m_dataLock };
         GST_DEBUG_OBJECT(m_transcoder.get(), "Transfering %zu encoded bytes, mimeType: %s", m_data.size(), mimeType.ascii().data());
         auto buffer = m_data.take();
-        completionHandler(WTFMove(buffer), mimeType, m_position);
+        completionHandler(WTFMove(buffer), mimeType, m_timeCode);
+        if (m_position.isValid())
+            m_timeCode = m_position.toDouble();
```

Comment:
If the position has been updated before the lock is acquired, shouldn't we update the timecode before calling the completion handler? i.e. having the if condition here before the completion handler call.

Reply:
Also, I'm wondering whether the data lock being held while the completion handler is called could be causing the timeouts. Maybe the lock can be released before calling it?

Reply:
Oh yes you're right, that's why the test is Pass/Timeout after this patch. Updating as you suggested makes it consistent Pass here :)